bip-tkhan-CCVS.mediawiki

```diff
@@ -0,0 +1,82 @@
+Author: t.khan <teekhan42@gmail.com>
+Comments-Summary: No comments yet.
+Comments-URI: TBD
+Status: Draft
+Type: Standards Track
+Created: 2017-02-10
+License: BSD-2
+Voting Address: 3CoFA3JiK5wxe9ze2HoDGDTmZvkE5Uuwh8  (just an example, don’t send to this!)
+</pre>
+
+==Abstract==
+
+Community Consensus Voting System (CCVS) will allow developers to measure support for BIPs prior to implementation.
+
+==Motivation==
+
+We currently have no way of measuring consensus for potential changes to the Bitcoin protocol. This is problematic for controversial changes such as the max block size limit. As a result, we have many proposed solutions but no clear direction.
+
+Also, due to our lack of ability to measure consensus, there is a general feeling among many in the community that developers aren’t listening to their concerns. This is a valid complaint, as it’s not possible to listen to thousands of voices all shouting different things in a crowded room—basically the situation in the Bitcoin community today.
+
+The CCVS will allow the general public, mining pools, and companies using Bitcoin to vote for their preferred BIP in a way that’s public and relatively difficult (expensive) to manipulate.
+
+The CCVS system will initially be used for measuring support for the various max block size limit related BIPs, so developers can move forward. In the future, the CCVS could be used for other issues, should the need arise.
+
+
+==Specification==
+
+Each competing BIP (currently 6 BIPs: #102 - #107) will be assigned a unique bitcoin address which is added to each header. Anyone who wants to vote would cast their ballot by sending a small amount (0.0001 btc) to their preferred BIP's address. Each transaction counts as 1 vote.
+
+Confirmed Vote Multiplier:
+Mining pools and companies using Bitcoin are allowed one confirmed vote each. A confirmed vote is worth 10,000x a regular vote.
+
+For example:
+
+Slush Pool casts a vote for its preferred BIP and then states publicly (on its blog) their vote and the transaction ID and emails the URL to the admin of this system. In the final tally, this vote will count as 10,000 votes.
+
+Coinbase, Antpool, BitPay, BitFury, etc., all do the same.
+
+Confirmed votes would be added to a new section in each respective BIP as a public record in the following format:
+<Company/Pool Name>	<Vote transaction ID>		<Confirmation URL>
+
+==Timing==
+
+Block #454,477 (~February 24, 2017): Final deadline for max block size BIP submissions for voting.
+
+Block #454,621 (~February 25, 2017): Voting officially begins.
+
+Block #455,629 (~March 4, 2017): Voting ends.
+
+==Implementation==
+
+Immediately: Announcement will be made on bitcoincore.org calling for any further max block size related BIP submissions by Block #454,477 (~February 24, 2017).
+
+Before Block #454,621: All max block size related BIPs are assigned a bitcoin address for voting. Voting begins. Announcement made on bitcoincore.org.
```

Comment:
Proposing voting on hardforks is in itself a hardfork, so this wouldn't be exclusively a Process BIP. Acceptance of this BIP would require approval from the entire Bitcoin community.